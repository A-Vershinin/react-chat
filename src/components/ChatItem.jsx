import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import moment from 'moment';
import { withStyles } from '@material-ui/core/styles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from './Avatar.jsx';

const styles = theme => ({
  activeItem: {
    backgroundColor: theme.palette.grey[200],
  },
});

const ChatItem = ({
  classes, title, chatId, active, disabled, createdAt,
}) => (
  <ListItem
    button
    component={Link}
    to={`/chat/${chatId}`}
    className={active ? classes.activeItem : ''}
    disabled={disabled}
  >
    <Avatar colorFrom={title}>{title}</Avatar>
    <ListItemText primary={title} secondary={moment(createdAt).fromNow()} />
  </ListItem>
);

ChatItem.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  title: PropTypes.string.isRequired,
  chatId: PropTypes.string.isRequired,
  active: PropTypes.bool,
  disabled: PropTypes.bool.isRequired,
  createdAt: PropTypes.string.isRequired,
};

ChatItem.defaultProps = {
  active: false,
};

export default withStyles(styles)(ChatItem);
