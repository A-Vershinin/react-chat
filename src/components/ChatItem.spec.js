/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatItem from './ChatItem.jsx';

jest.mock('./Avatar', () => () => 'Avatar');
// Return a fixed timestamp when moment().fromNow() is called
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const mockProps = {
  classes: {
    activeItem: 'lorem lorem',
  },
  title: 'some title',
  chatId: '12345',
  active: false,
  disabled: false,
  createdAt: '2018-03-16T10:53:23.200Z',
};

describe('<ChatItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatItem {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders when chatItem active', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatItem {...mockProps} active />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders when chatItem disabled', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <ChatItem {...mockProps} disabled />
              </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
