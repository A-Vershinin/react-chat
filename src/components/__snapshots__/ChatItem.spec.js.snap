// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatItem /> renders correctly 1`] = `
<a
  className="MuiButtonBase-root-63 MuiListItem-root-53 MuiListItem-default-56 MuiListItem-gutters-60 MuiListItem-button-61"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  Avatar
  <div
    className="MuiListItemText-root-66"
  >
    <span
      className="MuiTypography-root-72 MuiTypography-subheading-79 MuiListItemText-primary-69"
    >
      some title
    </span>
    <p
      className="MuiTypography-root-72 MuiTypography-body1-81 MuiTypography-colorTextSecondary-94 MuiListItemText-secondary-70"
    >
      2 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-96"
  />
</a>
`;

exports[`<ChatItem /> renders when chatItem active 1`] = `
<a
  className="MuiButtonBase-root-63 MuiListItem-root-53 MuiListItem-default-56 MuiListItem-gutters-60 MuiListItem-button-61 ChatItem-activeItem-52 lorem lorem"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  Avatar
  <div
    className="MuiListItemText-root-66"
  >
    <span
      className="MuiTypography-root-72 MuiTypography-subheading-79 MuiListItemText-primary-69"
    >
      some title
    </span>
    <p
      className="MuiTypography-root-72 MuiTypography-body1-81 MuiTypography-colorTextSecondary-94 MuiListItemText-secondary-70"
    >
      2 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-96"
  />
</a>
`;

exports[`<ChatItem /> renders when chatItem disabled 1`] = `
<a
  className="MuiButtonBase-root-63 MuiButtonBase-disabled-64 MuiListItem-root-53 MuiListItem-default-56 MuiListItem-gutters-60 MuiListItem-disabled-58 MuiListItem-button-61"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="-1"
>
  Avatar
  <div
    className="MuiListItemText-root-66"
  >
    <span
      className="MuiTypography-root-72 MuiTypography-subheading-79 MuiListItemText-primary-69"
    >
      some title
    </span>
    <p
      className="MuiTypography-root-72 MuiTypography-body1-81 MuiTypography-colorTextSecondary-94 MuiListItemText-secondary-70"
    >
      2 days ago
    </p>
  </div>
</a>
`;
